### PR TITLE
Dynamodb implementation of document node interface

### PR DESCRIPTION
**Pull request by jakobbraun**
Most of the files are quite similar. So propably it's enough to review for example `DynamodbString`, `DynamodbList`, `DynamodbObject`, their tests, `DynamodbNodeVisitor` and `DynamodbDocumentNodeFactory`.

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbObject.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.stream.Collectors;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.exasol.adapter.dynamodb.documentnode.DocumentNode;
+import com.exasol.adapter.dynamodb.documentnode.DocumentObject;
+
+/**
+ * This class represents a DynamoDB map value.
+ */
+public class DynamodbObject implements DocumentObject<DynamodbNodeVisitor> {
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbObject.java:13]: Sorry to pick on names today :) Here maybe you can also rename the class to `DynamodbObjectMap`?

Comment (morazow) [reply src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbObject.java:13]: Similar for other, if they extend DocumentArray, then rename to `*List`.

File: src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySet.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.exasol.adapter.dynamodb.documentnode.DocumentArray;
+import com.exasol.adapter.dynamodb.documentnode.DocumentNode;
+
+/**
+ * This class represents a DynamoDB binary set value.
+ */
+public class DynamodbBinarySet implements DocumentArray<DynamodbNodeVisitor> {
```
[the remark this answers is not in the document]
Comment (morazow) [reply src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySet.java:14]: From below class, AttributeValue wrapper I see that they use this convention of naming, `BS` -> `BinarySet`, then you can keep class names as it is also.

File: src/test/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySetTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        assertThat(visitor.wasVisited, equalTo(true));
+    }
+
+    private static class VisitationCheck implements DynamodbNodeVisitor {
+        boolean wasVisited = false;
+
+        @Override
+        public void visit(final DynamodbBinarySet value) {
+            this.wasVisited = true;
+        }
```
Comment (morazow) [inline src/test/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySetTest.java:51]:
```suggestion
        assertThat(visitor.visited, equalTo(true));
    }

    private static class VisitationCheck implements DynamodbNodeVisitor {
        boolean visited = false;

        @Override
        public void visit(final DynamodbBinarySet value) {
            this.visited = true;
        }
```

Comment (morazow) [reply src/test/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySetTest.java:51]: Please also change in other test classes.

File: src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySet.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySet.java:14]: As you mentioned the names follow the [DynamoDB type names](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_AttributeValue.html). To be consequent I did just renamed `DynamodbObject` to `DynamodbMap`.
I think using this names here fits better than generic ones as the Types are Dynamodb specific.

[the remark this answers is not in the document]
Comment (morazow) [reply src/main/java/com/exasol/adapter/dynamodb/documentnode/dynamodb/DynamodbBinarySet.java:14]: :+1:

**Pull request merged**